crates/matrix-sdk-ui/src/timeline/event_item/content/message.rs

```diff
@@ -139,6 +144,11 @@ impl Message {
         self.thread_root.is_some()
     }
 
+    /// Get the [`OwnedEventId`] of the root event of a thread if it exists.
+    pub fn thread_root(&self) -> Option<OwnedEventId> {
+        self.thread_root.clone()
```

Comment:
A slightly better API would consist in returning `Option<&EventId>` and let the caller decide whether they want to clone or not.

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/3942/commits/4b08774cb0142ea3553a7b1ab523ea1ade8eaa07